Spectrum/BetaQuest.cs

```diff
@@ -705,81 +364,444 @@ public static short[] GetVAR_Entrances()
+                { 0x0600, "Minuet of Forest Warp -> Sacred Forest Meadow" },
+                { 0x0604, "Serenade of Water Warp -> Lake Hylia" },
+                { 0x0608, "Phantom Ganon Boss Room Blue Warp -> SFM Forest Temple Entrance Ledge" },
+                { 0x060C, "Morpha Boss Room Blue Warp -> Lake Hylia" },
+                { 0x0610, "Twinrova Boss Room Blue Warp -> Desert Colossus From Spirit Lobby" },
+            };
+
+            // Entrances overridden by grottos
+            static readonly Dictionary<short, string> Grotto_Table = new()
+            {
+                { 0x003F, "HF Bomb Grotto near Market Entrance" },
+                { 0x036D, "HF Remote Bomb Grotto" },
+                { 0x00EE, "HF Open Grotto" },
+                { 0x0153, "Kak Open Grotto" },
+                { 0x0175, "ZR Open Grotto" },
+                { 0x01A5, "KF Storms Grotto" },
+                { 0x01AD, "LW Bomb Grotto" },
+                { 0x01ED, "DMT Storms Grotto" },
+                { 0x01F5, "DMC Bomb Grotto" },
+                { 0x024E, "ZD Storms Grotto" },
+                { 0x02B2, "GF Storms Grotto" },
+                { 0x02CE, "SFM Storms Grotto" },
+                { 0x02E2, "ZR Bomb Grotto" },
+                { 0x02E6, "HF Across River Bomb Grotto" },
+            };
         }
-        
+
         [SpectrumCommand(
-            Name = "beta_v",
-            Description = "sets version",
-            Cat = SpectrumCommand.Category.Proto,
+            Name = "bqspoiler",
+            Description = "Outputs Beta Quest entrance spoiler log",
+            Cat = SpectrumCommand.Category.Spectrum,
             Sup = SpectrumCommand.Supported.OoT)]
-        [SpectrumCommandSignature(
-            Sig = new Tokens[] { Tokens.LITERAL })]
-        private static void SetBetaQuestVersion(Arguments args)
+        private static void SpoilBetaQuest(Arguments args)
         {
             if (Options.Version != ORom.Build.N0)
                 return;
+            Ptr staticCtx = SPtr.New(0x11BA00).Deref();
+            Ptr shuffleTable = staticCtx.RelOff(BetaQuest.GetTBL_OFFSET());
 
-            string ver = (string)args[0];
+            short[] VAR_Entrances = BetaQuest.GetVAR_Entrances();
+            Dictionary<short, string> Entrance_Table = BetaQuest.GetEntrance_Table();
+            Dictionary<short, string> Grotto_Table = BetaQuest.GetGrotto_Table();
 
-            ver = $"V{ver.Replace('.', '_')}";
-            if (Enum.TryParse(ver, out BetaQuest.Ver result))
+            for (int i = 0; i < VAR_Entrances.Length; i++)
             {
-                BetaQuest.Version = result;
-                Console.WriteLine("Version updated.");
+                short entrance = VAR_Entrances[i];
+                string fromEntrance = Grotto_Table.ContainsKey(entrance) ? Grotto_Table[entrance] : Entrance_Table[entrance];
+
+                short read = shuffleTable.ReadInt16(i * 2);
+                string toEntrance = Entrance_Table[read];
```

Comment:
```suggestion
                string toEntrance = Grotto_Table.ContainsKey(index) ? Grotto_Table[index] : Entrance_Table[index];
```

Don't assume that `read` always maps to an Entrance_Table record